keylime/cloud_verifier_tornado.py

```
@@ -690,34 +695,108 @@ def delete(self) -> None:
+                states.SAVED,
+                states.FAILED,
+                states.TENANT_FAILED,
+                states.INVALID_QUOTE,
+            ):
+                # Agent is in a terminal state with no in-flight work — delete immediately
+                try:
+                    verifier_db_delete_agent(session, agent_id)
+                    web_util.echo_json_response(self.req_handler, 200, "Success")
+                    logger.info("DELETE (pull mode) returning 200 response for agent id: %s", agent_id)
+                except SQLAlchemyError as e:
+                    logger.error("SQLAlchemy Error deleting agent in pull mode: %s", e)
```

Ruff (0.15.9)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)